compiler/hash-storage/src/store/sequence/local.rs

```diff
@@ -0,0 +1,71 @@
+use std::marker::PhantomData;
+
+use super::SequenceStoreKey;
+
+/// An implementation of [SequenceStore] which doesn't require being shared
+/// across thread boundaries.
+pub struct LocalSequenceStore<Key, V> {
```

Comment:
How is this different from `IndexVec`?

Reply:
It can use the `SequenceStoreKey` API out of the box

Reply:
Okay, I see. Perhaps we should slowly start using this over `IndexVec` then

Reply:
Definitely in areas that lend them selves to using sequence stores